### PR TITLE
RC2 Branding Update

### PR DESCRIPTION
**Pull request by brthor**
Mostly changes from @Sridhar-MS and @piotrpMSFT for updating Branding and versioning where needed.

Fixes - #2833

### CODE REVIEW

File: TestAssets/TestPackages/dotnet-dependency-tool-invoker/project.json [hunk 1]
```diff
@@ -12,9 +12,9 @@
       "type": "platform",
       "version": "1.0.0-rc2-*"
     },
-    "Microsoft.DotNet.Cli.Utils": "1.0.0-*",
-    "Microsoft.DotNet.ProjectModel": "1.0.0-*",
-    "Microsoft.DotNet.Compiler.Common": "1.0.0-*"
+    "Microsoft.DotNet.Cli.Utils": "1.0.0-rc2-preview1-*",
```
Comment (piotrpMSFT) [inline TestAssets/TestPackages/dotnet-dependency-tool-invoker/project.json]: we need to replace `rc2-preview1-` with `preview1-`

Comment (brthor) [inline TestAssets/TestPackages/dotnet-dependency-tool-invoker/project.json]: good catch, it appears I did not press save after the find/replace 😄

Comment (brthor) [inline TestAssets/TestPackages/dotnet-dependency-tool-invoker/project.json]: done.

File: branchinfo.txt [hunk 2]
```diff
@@ -4,5 +4,5 @@
 MAJOR_VERSION=1
 MINOR_VERSION=0
 PATCH_VERSION=0
-RELEASE_SUFFIX=rc2
+RELEASE_SUFFIX=rc2-preview1
```
Comment (brthor) [inline branchinfo.txt]: @piotrpMSFT Having some trouble updating this file, IIRC you did some git magic here, is there something special I need to do to update it?

File: packaging/windows/clisdk/registrykeys.wxs [hunk 3]
```diff
@@ -8,6 +8,11 @@
           <RegistryValue Action="write" Name="$(var.NugetVersion)" Type="integer" Value="1" KeyPath="yes"/>
         </RegistryKey>
       </Component>
+      <Component Id="SetupRegistry_x86_RC2_Compat_Key" Directory="TARGETDIR" Win64="no">
+        <RegistryKey Root="HKLM" Key="SOFTWARE\dotnet\Setup\InstalledVersions\$(var.Platform)\sdk">
+          <RegistryValue Action="write" Name="1.0.0-rc2-preview1" Type="integer" Value="1" KeyPath="yes"/>
```
Comment (brthor) [inline packaging/windows/clisdk/registrykeys.wxs]: @Sridhar-MS is this preview1 appropriate here?

Comment (Sridhar-MS) [inline packaging/windows/clisdk/registrykeys.wxs]: Yup. This needs to be updated.

File: TestAssets/DesktopTestProjects/DesktopAppWhichCallsDotnet/project.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
   "version": "1.0.0-*",
   "dependencies": {
-    "Microsoft.DotNet.Cli.Utils": "1.0.0-*"
+    "Microsoft.DotNet.Cli.Utils": "1.0.0-rc2-*"
   },
```
Comment (Sridhar-MS) [inline TestAssets/DesktopTestProjects/DesktopAppWhichCallsDotnet/project.json]: All our test references need to be changed to `1.0.0-preview1`

Comment (brthor) [inline TestAssets/DesktopTestProjects/DesktopAppWhichCallsDotnet/project.json]: changing this

Comment (brthor) [inline TestAssets/DesktopTestProjects/DesktopAppWhichCallsDotnet/project.json]: using this regex: `\"Microsoft\.DotNet\.Cli\.Utils\"\s*:\s*\"1\.0\.0-[^p]`

This Is the only one I'm finding that isn't right

File: scripts/dotnet-cli-build/CompileTargets.cs [hunk 5]
```diff
@@ -627,7 +627,9 @@ private static BuildTargetResult CompileCliSdk(BuildTargetContext c, DotNetCli d
                     outputDir,
                     "--configuration",
                     configuration,
-                    Path.Combine(c.BuildContext.BuildDirectory, "src", project))
+                    Path.Combine(c.BuildContext.BuildDirectory, "src", project),
+                    "--version-suffix",
+                    buildVersion.VersionSuffix)
```
Comment (Sridhar-MS) [inline scripts/dotnet-cli-build/CompileTargets.cs]: @brthor  Why is this required?

File: TestAssets/DesktopTestProjects/DesktopAppWhichCallsDotnet/project.json (hunk 4 above)
Comment (brthor) [inline TestAssets/DesktopTestProjects/DesktopAppWhichCallsDotnet/project.json]: I'm going to step through the other nugget packages and make sure everything is right too

File: scripts/dotnet-cli-build/Utils/Monikers.cs [hunk 6]
```diff
@@ -9,9 +9,9 @@ namespace Microsoft.DotNet.Cli.Build
     public class Monikers
     {
         public const string SharedFrameworkName = "Microsoft.NETCore.App";
-        public const string CLISdkBrandName = "Microsoft .NET Core SDK";
-        public const string SharedFxBrandName = "Microsoft .NET Core";
-        public const string SharedHostBrandName = "Microsoft .NET Core Host";
+        public const string CLISdkBrandName = "Microsoft .NET Core 1.0 RC2 - SDK Preview 1";
```
Comment (Sridhar-MS) [inline scripts/dotnet-cli-build/Utils/Monikers.cs]: @piotrpMSFT What is the correct SDK brand name now?

File: scripts/dotnet-cli-build/CompileTargets.cs (hunk 5 above)
Comment (brthor) [inline scripts/dotnet-cli-build/CompileTargets.cs]: I didn't add this, @piotrpMSFT ?

File: TestAssets/TestPackages/dotnet-dependency-context-test/project.json [hunk 7]
```diff
@@ -1,5 +1,5 @@
 {
-  "version": "1.0.0-*",
+  "version": "1.0.0",
```
Comment (eerhardt) [inline TestAssets/TestPackages/dotnet-dependency-context-test/project.json]: why is this needed?

File: src/Microsoft.DotNet.ProjectModel/project.json [hunk 8]
```diff
@@ -1,11 +1,11 @@
 {
-  "version": "1.0.0-*",
+  "version": "1.0.0-rc2-*",
```
Comment (eerhardt) [inline src/Microsoft.DotNet.ProjectModel/project.json]: I don't understand why some of our assemblies are rc2 and some are preview1.  Is the reasoning written down somewhere?

File: TestAssets/TestPackages/dotnet-dependency-context-test/project.json (hunk 7 above)
Comment (brthor) [inline TestAssets/TestPackages/dotnet-dependency-context-test/project.json]: same question

File: src/Microsoft.DotNet.ProjectModel/project.json (hunk 8 above)
Comment (brthor) [inline src/Microsoft.DotNet.ProjectModel/project.json]: @piotrpMSFT has this context. Last minute branding changes

File: scripts/dotnet-cli-build/Utils/HostVersion.cs [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+        // Host versions are independent of CLI versions. Moreover, these version numbers
+        // are baked into the binary and is used to look up a serviced binary replacement.
+        //
+
+        //
+        // Latest hosts for production of nupkgs.
+        //
+
+        // Version constants without suffix
+        public override int Major => 1;
```
Comment (eerhardt) [inline scripts/dotnet-cli-build/Utils/HostVersion.cs]: Instead of an override, would it make more sense to just set these properties in the constructor?

File: scripts/dotnet-cli-build/Utils/HostVersion.cs [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+        {
+            { "Microsoft.NETCore.DotNetHost", LatestHostVersion },
+            { "Microsoft.NETCore.DotNetHostResolver", LatestHostFxrVersion },
+            { "Microsoft.NETCore.DotNetHostPolicy", LatestHostPolicyVersion }
+        };
+
+        //
+        // Locked muxer for consumption in CLI.
+        //
+        public bool IsLocked = false; // Set this variable to toggle muxer locking.
```
Comment (eerhardt) [inline scripts/dotnet-cli-build/Utils/HostVersion.cs]: This appears to never be set to true.

Comment (Sridhar-MS) [inline scripts/dotnet-cli-build/Utils/HostVersion.cs]: This will be set to true when we want to lock the muxer and the hostfxr packages.

File: scripts/dotnet-cli-build/Utils/BuildVersion.cs [hunk 11]
```diff
@@ -5,7 +5,7 @@ namespace Microsoft.DotNet.Cli.Build
     public class BuildVersion : Version
     {
         public string SimpleVersion => $"{Major}.{Minor}.{Patch}.{CommitCountString}";
-        public string VersionSuffix => $"{CommitCountString}";
+        public string VersionSuffix => $"{ReleaseSuffix}-{CommitCountString}";
```
Comment (piotrpMSFT) [inline scripts/dotnet-cli-build/Utils/BuildVersion.cs]: This should not be here. The Release Suffix is already in the nupkg versions, so this doubles it up. We don't want to get nupkgs with -preview1-preview1-{version}.

File: scripts/dotnet-cli-build/Utils/BuildVersion.cs [hunk 12]
```diff
@@ -2,82 +2,12 @@
...
-        public int Minor { get; set; }
-        public int Patch { get; set; }
-        public int CommitCount { get; set; }
-        public string CommitCountString => CommitCount.ToString("000000");
-        public string ReleaseSuffix { get; set; }
-
         public string SimpleVersion => $"{Major}.{Minor}.{Patch}.{CommitCountString}";
-        public string VersionSuffix => $"{ReleaseSuffix}-{CommitCountString}";
+        public string VersionSuffix => $"{CommitCountString}";
         public string NuGetVersion => $"{Major}.{Minor}.{Patch}-{VersionSuffix}";
```
Comment (Sridhar-MS) [inline scripts/dotnet-cli-build/Utils/BuildVersion.cs]: `VersionSuffix` is used in multiple places including packing our test packages https://github.com/dotnet/cli/blob/rel/1.0.0/scripts/dotnet-cli-build/TestTargets.cs#L204. So for test packages the nuget version will look like `1.0.0-002668`.

Also now the `NuGetVersion` which used everywhere in all the installers will now be just `1.0.0-002668` instead of `1.0.0-preview1-002668`

Comment (Sridhar-MS) [inline scripts/dotnet-cli-build/Utils/BuildVersion.cs]: Also @brthor made changes to our [package targets](https://github.com/dotnet/cli/pull/2877/files#diff-cf3b35d6a17182b432693f932941e59cR217) to use the commit count as the version suffix for the nuget packages. But this fix was not made for the pack command in the [test targets](https://github.com/dotnet/cli/blob/rel/1.0.0/scripts/dotnet-cli-build/TestTargets.cs#L165).

Comment (brthor) [inline scripts/dotnet-cli-build/Utils/BuildVersion.cs]: @Sridhar-MS are you going to change this back? I thought that's what we talked about

Comment (brthor) [inline scripts/dotnet-cli-build/Utils/BuildVersion.cs]: I'll spend some time and verify that we use commit count anywhere we do pack

File: scripts/dotnet-cli-build/CompileTargets.cs (hunk 5 above)
Comment (brthor) [inline scripts/dotnet-cli-build/CompileTargets.cs]: going to remove this

File: TestAssets/TestPackages/dotnet-dependency-context-test/project.json (hunk 7 above)
Comment (brthor) [inline TestAssets/TestPackages/dotnet-dependency-context-test/project.json]: going to add this back

File: scripts/dotnet-cli-build/Utils/BuildVersion.cs (hunk 12 above)
Comment (brthor) [inline scripts/dotnet-cli-build/Utils/BuildVersion.cs]: I'll go ahead and add this back and track down the pack usages @Sridhar-MS

File: scripts/dotnet-cli-build/Utils/BuildVersion.cs [hunk 13]
```diff
@@ -2,82 +2,12 @@
 
 namespace Microsoft.DotNet.Cli.Build
 {
-    public class BuildVersion
+    public class BuildVersion : Version
```
Comment (brthor) [inline scripts/dotnet-cli-build/Utils/BuildVersion.cs]: Why do we need this layer of inheritance here? 
@Sridhar-MS @piotrpMSFT

File: scripts/dotnet-cli-build/Utils/HostVersion.cs (hunk 9 above)
Comment (brthor) [inline scripts/dotnet-cli-build/Utils/HostVersion.cs]: Why are we representing the different versions in our build with inheritance like this. 
I mean it works, but why not put all the versions in one place similar to how it was done previously? It at least makes these things easier to change.

File: scripts/dotnet-cli-build/CompileTargets.cs (hunk 5 above)
Comment (piotrpMSFT) [inline scripts/dotnet-cli-build/CompileTargets.cs]: Look @line 622. it suggests we add this. Since we need for version suffix to be understood, and it seemed not to be, added this optimistically.

File: scripts/dotnet-cli-build/Utils/BuildVersion.cs (hunk 11 above)
Comment (brthor) [inline scripts/dotnet-cli-build/Utils/BuildVersion.cs]: We are using CommitCountString as the version suffix for the nupkgs now.

**Comment by brthor** [comment]
http://dotnet-ci.cloudapp.net/job/dotnet_cli/job/rel_1.0.0/job/release_rhel7.2_x64_prtest/1778/

Rhel one test failing, going to kick back off in case it's transitive, saving the link so we can look at this later.

cc @cdmihai This is an incremental build test it looks like

**Comment by brthor** [comment]
@dotnet-bot test RHEL7.2 x64 Release Build please

**Comment by brthor** [comment]
@dotnet-bot test OSX x64 Release Build please

**Comment by cdmihai** [comment]
Regarding the failed test, the only way the current change can affect it (as far as I see) is if the contents of the .version file changed somehow (changed, moved, deleted): https://github.com/dotnet/cli/blob/rel/1.0.0/scripts/dotnet-cli-build/CompileTargets.cs#L584

**Comment by brthor** [comment]
@cdmihai I don't see how that could happen unless Jenkins workspaces conflicted. 

Seems pretty unlikely, but it hasn't failed again. We'll keep an eye out.

**Comment by piotrpMSFT** [comment]
Thanks for all the hard work on this one, @brthor @Sridhar-MS @eerhardt @joelverhagen
